src/service/standard-actions-impl.js

```diff
@@ -103,6 +105,42 @@ export class StandardActions {
     );
   }
 
+  /**
+   * Handles initiliazing the theme mode.
+   *
+   * This methode needs to be called on page load to set the `amp-dark-mode`
+   * class on the body if the user prefers the dark mode.
+   */
+  initThemeMode_() {
+    if (this.prefersDarkMode_()) {
+      this.ampdoc.waitForBodyOpen().then((body) => {
+        const darkModeClass =
+          body.getAttribute('prefers-dark-mode-class') || 'amp-dark-mode';
```

Comment:
We can use a `data-*` attribute to avoid any changes to validator.

```suggestion
          body.getAttribute('data-prefers-dark-mode-class') || 'amp-dark-mode';
```